rest/tests/test_search.py

```diff
@@ -548,14 +524,153 @@ def test_selected_facets(self):
+        won't mix with resources that are actually missing terms.
+        """
+        # Create 'empty' term. Term is currently not assigned to anything.
+        vocab_dict = dict(self.DEFAULT_VOCAB_DICT)
+        vocab_dict['learning_resource_types'] = [
+            self.resource.learning_resource_type.name
+        ]
+        vocab_slug = self.create_vocabulary(self.repo.slug, vocab_dict)['slug']
+        term_slug = self.create_term(self.repo.slug, vocab_slug, {
+            "label": "empty",
+            "weight": 4
+        })['slug']
+        vocab_key = "vocab_{slug}".format(slug=vocab_slug)
+
+        # There is one resource and it is missing a term.
+        # Test that a missing search will get one result.
+        results = self.get_results(
+            selected_facets=["_missing_:{v}_exact".format(v=vocab_key)]
+        )
+        self.assertEqual(results['count'], 1)
+        self.assertEqual(
+            results['facet_counts'][vocab_key]['facet']['missing_count'], 1)
+
+        # A search for 'empty' should get zero results.
+        results = self.get_results(
+            selected_facets=["{v}_exact:{t}".format(
+                v=vocab_key,
+                t=term_slug
+            )]
+        )
+        self.assertEqual(results['count'], 0)
+        self.assertEqual(
+            results['facet_counts'][vocab_key]['facet']['missing_count'], 0)
+
+        # Assign term to learning resource
+        self.patch_learning_resource(self.repo.slug, self.resource.id, {
+            "terms": [term_slug]
+        })
+
+        # Do missing search again.
+        results = self.get_results(
+            selected_facets=["_missing_:{v}_exact".format(v=vocab_key)]
+        )
+        self.assertEqual(results['count'], 0)
+        self.assertEqual(
+            results['facet_counts'][vocab_key]['facet']['missing_count'],
+            0
+        )
+
+        # Do search for term.
+        results = self.get_results(
+            selected_facets=["{v}_exact:{t}".format(
+                v=vocab_key,
+                t=term_slug
+            )]
+        )
+        self.assertEqual(results['count'], 1)
+        self.assertEqual(
+            results['facet_counts'][vocab_key]['facet']['missing_count'],
+            0
+        )
+
+    def test_name_collision(self):
+        """
+        Test that collisions won't happen for terms. Vocabularies can't
+        have the same name so we don't test that here.
+        """
+        vocab1 = Vocabulary.objects.create(
+            repository=self.repo,
+            required=False,
+            weight=1,
+            name='voc with spaces 1',
+            multi_terms=True
+        )
+        vocab2 = Vocabulary.objects.create(
+            repository=self.repo,
+            required=False,
+            weight=1,
+            name='voc with spaces 2',
+            multi_terms=True
+        )
+        vocab1.learning_resource_types = [self.resource.learning_resource_type]
```

Comment:
Do these two lines work without calling the `.add()` method after adding to the M2M table?